### PR TITLE
Extend whitelist in global initialization checker

### PR DESCRIPTION
**Pull request by EnzeXing**
This PR extends the whitelist in global object initialization checker in order to suppress initialization irrelevance errors in Scala2LibraryTasty.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/init/Objects.scala [hunk 1]
```diff
@@ -68,11 +69,21 @@ import dotty.tools.dotc.core.Flags.AbstractOrTrait
...
+  val immutableVector: Symbol = requiredModule("scala.collection.immutable.Vector")
+  val Vector_EmptyIterator: Symbol = immutableVector.requiredValue("emptyIterator")
+  val immutableMapNode: Symbol = requiredModule("scala.collection.immutable.MapNode")
+  val MapNode_EmptyMapNode: Symbol = immutableMapNode.requiredValue("EmptyMapNode")
+  val immutableHashMap: Symbol = requiredModule("scala.collection.immutable.HashMap")
+  val HashMap_EmptyMap: Symbol = immutableHashMap.requiredValue("EmptyMap")
+  val immutableLazyList: Symbol = requiredModule("scala.collection.immutable.LazyList")
+  val LazyList_empty: Symbol = immutableLazyList.requiredValue("_empty")
+
+  val whiteList: Set[Symbol] = Set()
```
Comment (olhotak) [inline compiler/src/dotty/tools/dotc/transform/init/Objects.scala:86]: Did you mean it to be empty?

File: compiler/src/dotty/tools/dotc/transform/init/Objects.scala [hunk 2]
```diff
@@ -821,6 +832,7 @@ class Objects(using Context @constructorOnly):
               errorReadOtherStaticObject(State.currentObject, addr)
               Bottom
           else if ref.isObjectRef && ref.klass.hasSource then
+            println(s"Uninitialized field Position 2, ref = $ref, target = $target")
```
Comment (olhotak) [inline compiler/src/dotty/tools/dotc/transform/init/Objects.scala:835]: You probably didn't intend to leave these printlns in.

File: compiler/src/dotty/tools/dotc/transform/init/Objects.scala (hunk 1 above)
Comment (EnzeXing) [reply compiler/src/dotty/tools/dotc/transform/init/Objects.scala]: Sorry, I forgot to push this. It has been corrected.

File: compiler/src/dotty/tools/dotc/transform/init/Objects.scala (hunk 2 above)
Comment (EnzeXing) [reply compiler/src/dotty/tools/dotc/transform/init/Objects.scala]: It has been corrected

**Review by olhotak — APPROVED** [review]
LGTM

@liufengyun would you also like to have a look?

**Review by liufengyun — APPROVED** [review]
LGTM, thank you @EnzeXing !

**Pull request merged**